src/pages/Login.js

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link, useNavigate } from "react-router-dom";
+import { login } from "../features/authSlice";
+import { useLocation } from "react-router-dom";
+
+export const Login = () => {
+  const [form, setForm] = useState({ username: "", password: "" });
+  const dispatch = useDispatch();
+  const { user } = useSelector((state) => state.auth);
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const handleFormChange = (e) => {
+    const { name, value } = e.target;
+    setForm({ ...form, [name]: value });
+  };
+
+  const handleLogin = (e) => {
+    e.preventDefault();
+    dispatch(login(form));
+  };
+  useEffect(() => {
+    if (user) {
+      navigate(location.state?.from?.pathname ?? "/", { replace: true });
+    }
+  }, [navigate, user, location]);
```

Comment:
Asking this for confirmation , you are writing this because you want the authenticated user to avoid accessing the login route.

Reply:
yes